jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildImageTask.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.image.json.BuildableManifestTemplate;
+import com.google.common.base.Preconditions;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.TaskAction;
+
+/** Builds a container image. */
+public class BuildImageTask extends DefaultTask {
+
+  /** Linked extension that configures this task. Must be set before the task is executed. */
+  @Nullable private JibExtension extension;
```

Comment:
Oh this null check really makes our code weird huh. We can't guarantee set Extension is called with a non-null value before anything else is called?

It feels like the null checker is overlapping with gradle's own checking mechanisms.

Reply:
Yeah, it is overlapping. I changed `setExtension` to `applyExtension` so it is applying the params directly.